src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,1591 @@
+			}
+			switch (val)
+			{
+				case SYSTEMOUTPUTS::SUPPLY_AIR_TEMP:  
+					if (MassFlowDependentDenominator==0)
+					{
+						averagedVal = StandbyMode->SupplyAirTemperature;
+					}
+					else averagedVal = averagedVal/ MassFlowDependentDenominator; break;
+				case SYSTEMOUTPUTS::SUPPLY_AIR_HR:  
+					if (MassFlowDependentDenominator == 0)
+					{
+						averagedVal = StandbyMode->SupplyAirW;
+					}
+					else averagedVal = averagedVal / MassFlowDependentDenominator; break;
+				case SYSTEMOUTPUTS::MIXED_AIR_TEMP:  
+					if (MassFlowDependentDenominator == 0)
+					{
+						averagedVal = StandbyMode->Mixed_Air_Temperature;
+					}
+					else averagedVal = averagedVal / MassFlowDependentDenominator; break;
+				case SYSTEMOUTPUTS::MIXED_AIR_HR: 
+					if (MassFlowDependentDenominator == 0)
+					{
+						averagedVal = StandbyMode->Mixed_Air_W;
+					}
+					else averagedVal = averagedVal / MassFlowDependentDenominator; break;
+			}
+			return averagedVal;
+		}
+		double Model::CalculatePartRuntimeFraction(double Mvent, double MinOA_Msa, double RequestedConditioningLoad,double SensibleRoomORZone, double RequestedLatentRoomORZone, double ZoneDehumidificationLoad, double requestedhumidificationload, double humidification)
+		{
+			double PLLatetRatio, PLVentRatio, PLSensibleRatio, PartRuntimeFraction;
+			PLLatetRatio = PLVentRatio = PLSensibleRatio = 0;
+
+			if (Mvent > 0)   PLVentRatio = MinOA_Msa / Mvent;
+			PartRuntimeFraction = 0; //PLVentRatio; tempory fix to deal with fact we haven't yet implimented a ventilation only backup mode.
+			if (SensibleRoomORZone != 0)   PLSensibleRatio = abs(RequestedConditioningLoad) / abs(SensibleRoomORZone);
+			if (PLSensibleRatio > PartRuntimeFraction) PartRuntimeFraction = PLSensibleRatio;
+			if (ZoneDehumidificationLoad > 0)  PLLatetRatio = RequestedLatentRoomORZone / ZoneDehumidificationLoad;
+			if (PLLatetRatio > PartRuntimeFraction) PartRuntimeFraction = PLLatetRatio;
+
+			return PartRuntimeFraction;
+		}
+
+		int Model::SetOperatingSetting(CStepInputs StepIns)
+		{
+
+
+			double Requestedlatent = 0; //needs to come from EP
+			bool DidWeMeetLoad = false;
+			bool DidWeMeetHumidificaiton = false;
+			bool DidWePartlyMeetLoad = false;
+			using General::RoundSigDigits;
+			int modenumber = 0;
+			int point_number = 0;
+			double MsaRatio = 0;
+			double OSAF = 0;
+			double Msa = 0;
+			double Mvent = 0;
+			double pOptimal_RunFractionTotalFuel = IMPLAUSIBLE_POWER;
+			double EIR, ElectricalPower, SHR, Tma, Wma, Hsa, Hma, TotalSystem;
+			//mode_optimal_power = 0;
+			double PreviousMaxiumConditioningOutput = 0;
+			double PreviousMaxiumHumidOrDehumidOutput = 0;
+			
+			
+			if (StepIns.RHosa > 1) { ShowSevereError("Unitary hybrid system error, required RH 0-1"); } //should be fractional 
+			if (StepIns.RHra > 1) { ShowSevereError("Unitary hybrid system error, required RH 0-1"); } //should be fractional 
```

Comment:
Severe errors lead directly to fatal errors.  If this is actually severe, such that the simulation must stop, you need a ShowFatalError after this.  If this is not a fatal error, and the simulation can continue, it needs to throw a warning instead.